libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/RX/NetworkInterface.c

```diff
@@ -286,11 +233,10 @@ static void prvEMACDeferredInterruptHandlerTask( void * pvParameters )
 
     for( ; ; )
     {
-        #if ( ipconfigHAS_PRINTF != 0 )
-            {
-                prvMonitorResources();
-            }
-        #endif /* ipconfigHAS_PRINTF != 0 ) */
+		/* Call a function that monitors resources: the amount of free network
+		buffers and the amount of free space on the heap.  See FreeRTOS_IP.c
+		for more detailed comments. */
+		vPrintResourceStats();
```

Comment:
This function will be undefined if `ipconfigHAS_PRINTF == 0` or is not defined (See FreeRTOS_IP.c:2355).  I think this should be:

```c
         #if ( ipconfigHAS_PRINTF != 0 )
 		vPrintResourceStats(); 
         #endif /* ipconfigHAS_PRINTF != 0 ) */ 
```

Reply:
It is defined in FreeRTOS_IP.h on line 310 above as:
```
#if ( ipconfigHAS_PRINTF != 0 )
	extern void vPrintResourceStats( void );
#else
	#define vPrintResourceStats()	do {} while( ipFALSE_BOOL )
#endif
```

Reply:
Cobus suggested to do it [like this](https://github.com/aws/amazon-freertos/pull/1825#pullrequestreview-372336437).
My original solution was like Gary's, using the `#ifdef` at the place where the function is called.
~~~c
#if ( ipconfigHAS_PRINTF != 0 )
    vPrintResourceStats(); 
#endif /* ipconfigHAS_PRINTF != 0 ) */ 
~~~
The advantage is that it is visible on what condition the function will be called.